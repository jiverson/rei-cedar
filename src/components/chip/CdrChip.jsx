import clsx from 'clsx';
import style from './styles/CdrChip.scss';
import modifier from '../../mixins/modifier';
import propValidator from '../../utils/propValidator';

export default {
  name: 'CdrChip',
  mixins: [modifier],
  props: {
    modifier: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['default', 'emphasis'],
      ),
      default: 'default',
    },
  },
  data() {
    return {
      style,
      baseClass: 'cdr-chip',
    };
  },
  render() {
    return (
      <button
        class={clsx(this.style[this.baseClass], this.modifierClass)}
        {...{ on: this.$listeners }}
      >
        { this.$slots['icon-left'] && (
          <span class={this.style['cdr-chip__icon-left']}>
            { this.$slots['icon-left'] }
          </span>
        )}
        { this.$slots.default }
        { this.$slots['icon-right'] && (
          <span class={this.style['cdr-chip__icon-right']}>
            { this.$slots['icon-right'] }
          </span>
        )}
      </button>
    );
  },
};
