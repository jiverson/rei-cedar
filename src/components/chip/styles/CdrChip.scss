@import '../../../css/settings/index.scss';
@import './vars/CdrChip.vars.scss';

.cdr-chip {
  @include cdr-chip-base-mixin;
}
