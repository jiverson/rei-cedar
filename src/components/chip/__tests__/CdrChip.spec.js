import { shallowMount } from '@vue/test-utils';
import CdrChip from 'componentdir/chip/CdrChip';

describe('CdrChip', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(CdrChip, {
      slots: {
        default: 'tortilla chip'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  // it('has a failing test by default so you remember to do them', () => {
  //   expect(false).toBe(true);
  // });
});
