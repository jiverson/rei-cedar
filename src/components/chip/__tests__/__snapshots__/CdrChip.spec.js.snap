// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrChip matches snapshot 1`] = `
<button
  class="cdr-chip"
>
  tortilla chip
</button>
`;
